src/providers/dart_rename_provider.ts

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+import { window, workspace, RenameProvider, WorkspaceEdit, TextDocument, Position, CancellationToken, Uri, TextEdit, Range } from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+import * as as from "../analysis/analysis_server_types";
+
+export class DartRenameProvider implements RenameProvider {	
+	private analyzer: Analyzer;
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+	
+	provideRenameEdits(document: TextDocument, position: Position, newName: string, token: CancellationToken): Thenable<WorkspaceEdit> {		
+		return workspace.saveAll(false).then(() => {
+			return this.doRename(document, position, newName, token);
+		});
+	}
+
+	private doRename(document: TextDocument, position: Position, newName: string, token: CancellationToken): Thenable<WorkspaceEdit> {
+        return new Promise<WorkspaceEdit>((resolve, reject) => {
+			this.analyzer.editGetRefactoring({
+				kind: "RENAME",
+				file: document.fileName,
+				offset: document.offsetAt(position),
+				length: position.character,
```

Comment:
The edits will have had their own ranges, so I guess the AS server just used the first symbol it found at the offset (maybe the length doesn't matter for this refactoring; it would be more important for things like "extract method" if such a thing exists).